Apex 45:  Datadog support

antoinegoutagny opened the pull request:
Add support for datadog transport. Uses aws secret for datadog api key.

antoinegoutagny committed:
Add simple test suite

antoinegoutagny committed:
APEX-45: Add dependencies

- aws-sdk for secrets manager
- datadog transport plugin for winston

antoinegoutagny committed:
APEX-45: support datadog transport

retrieves api key from aws secret

antoinegoutagny committed:
add datadog to local test

antoinegoutagny committed:
APEX-45: bump version number, doc

Update doc.

antoinegoutagny committed:
Use json format for console output

best way to display metadata props along with message.
+ add forgotten data sent along error.

antoinegoutagny commented:
TODO: Datadog setting is done asynchronously, with no way to know when it's done. Logging done before that will be lost for datadog. I would like the logger to expose a promise property that resolves when all transports have been set up. In an app, we can await it before doing anything, to make sure we don't lose any info.

agiletkiewicz commented:
Thank you for this Antoine! I am excited to test it out when we get back on Tuesday. One nit before then - feels like a big enough change to bump a major version.

antoinegoutagny commented:
> One nit before then - feels like a big enough change to bump a major version.

We try to follow semantic versionning. As this is just adding datadog support without any breaking change, I see no need to bump the major version.

antoinegoutagny committed:
Add ready property to Logger

ready instance property (readonly) returns a promise that resolves when all transports are ready. Or rejects if any transport setup fails.

antoinegoutagny committed:
ready promise returns init duration
